core/src/main/java/com/graphhopper/routing/weighting/FastestWeighting.java

```diff
@@ -62,7 +62,7 @@ public FastestWeighting(FlagEncoder encoder, PMap map, TurnCostProvider turnCost
         super(encoder.getAccessEnc(), encoder.getAverageSpeedEnc(), turnCostProvider);
         headingPenalty = map.getDouble(Routing.HEADING_PENALTY, Routing.DEFAULT_HEADING_PENALTY);
         headingPenaltyMillis = Math.round(headingPenalty * 1000);
-        maxSpeed = encoder.getMaxSpeed() / SPEED_CONV;
+        maxSpeed = (speedEnc.getRealMax() < 0 ? speedEnc.getMaxDecimal() : speedEnc.getRealMax()) / SPEED_CONV;
```

Comment:
Can we move this as a protected function into AbstractWeighting?

Reply:
I moved it to static `Weighting#getUpperSpeedBound` here: 5c8beb27ecea0c3497a6a0bdaa6cdf04038a1e82
Or shall we move it to `EncodedValue#getUpperBound(EncodedValue)`? Or even do this internally (return the storable max or the real max depending on whether or not it was set already)? Especially since it is only an approximation for the max value now that could be even better?

Reply:
Yes! :D ca06878bc48cd382fdfb12cb0920b8f8e3c01090